src/unitxt/artifact.py

```diff
@@ -456,20 +509,27 @@ def fetch_artifact(artifact_rep) -> Tuple[Artifact, Union[AbstractCatalog, None]
     (5) Otherwise, check that the artifact representation is a dictionary and build an Artifact object from it.
     """
     if isinstance(artifact_rep, Artifact):
+        if isinstance(artifact_rep, ArtifactLink):
```

Comment:
I'm not sure about this code.  If ww do fetch_artificant(ArtifiactLink("cards.sst2")), then we don't expect to get "cards.sst2", bur rather the real Card from "cards.sst2".  
Can you add a check a test to see this is what we get?

Reply:
Yes. Looks right.